From dea74e741a8a0ce3933aca44498bf810298076e7 Mon Sep 17 00:00:00 2001
From: vamsikrishnak <vamsikrishnak@ami.com>
Date: Tue, 26 Dec 2023 13:45:19 +0530
Subject: [PATCH] Fixed Bmcweb Crash when VirtualMedia disconnected

When we call a null weakptr, it will cause a crash.
Add nullptr check for weakptr can avoid this situation.

Tested:
bmcweb.service did not experience core-dump.

Signed-off-by: vamsikrishnak <vamsikrishnak@ami.com>
---
 http/websocket.hpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/http/websocket.hpp b/http/websocket.hpp
index c7d825f..9a6ef6d 100644
--- a/http/websocket.hpp
+++ b/http/websocket.hpp
@@ -165,6 +165,11 @@ class ConnectionImpl : public Connection
                        [weak(weak_from_this()), onDone{std::move(onDone)}](
                            const boost::beast::error_code& ec, size_t) {
             std::shared_ptr<Connection> self = weak.lock();
+            if (!self)
+            {
+                BMCWEB_LOG_ERROR("Connection went away");
+                return;
+            }
 
             // Call the done handler regardless of whether we
             // errored, but before we close things out
-- 
2.9.5

